lib/tpm2_policy.c

```diff
@@ -200,6 +200,81 @@ bool tpm2_policy_build_pcr(TSS2_SYS_CONTEXT *sapi_context,
     return result;
 }
 
+bool tpm2_policy_build_policyauthorize(TSS2_SYS_CONTEXT *sapi_context,
+    tpm2_session *policy_session, const char *to_authorize_policy_file,
+    const char *to_authorize_policy_qualifier_file,
+    const char *verifying_public_key_name_file,
+    const char *verification_ticket_file) {
+
+    unsigned long file_size = 0;
+
+    if (!files_get_file_size_path(to_authorize_policy_file, &file_size)) {
+        LOG_ERR("Could not retrieve file size of the policy to authorize");
+        return false;
+    }
+    TPM2B_DIGEST approvedPolicy = {
+        .size = (uint16_t)file_size
```

Comment:
No need to do this. Why don't you make sure file size is not bigger then 16 bytes...
```C
if (file_size & ~0xFFFF) {
    // too big error
}
```

Reply:
@williamcroberts, this check is telling that there was no qualifier file given.